### PR TITLE
CR-1090772 xbutil dump should not show information for soft kernel in compute_unit

### PR DESCRIPTION
**Pull request by AShivangi**
**Work Done:** separated out "ps_compute_units" from "compute_units"

**Terminologies:**
- _Compute units_: All CUs
- _PL compute units_: Programmable Logic CUs
- _PS compute units_: Processing System CUs

**Commands affected:**
- xbutil dump
- xbutil top
- xbutil2 examine -r compute_units (-f json)

**Sample outputs for the above-mentioned commands:**
```
$ xbutil dump -d 1
...
"compute_unit": {
            "0": {
                "name": "decoder:decoder_1",
                "base_address": "25161984",
                "usage": "0",
                "status": "(--)"
            },
            "1": {
                "name": "verify:verify_1",
                "base_address": "20971520",
                "usage": "0",
                "status": "(IDLE)"
            }
        },
        "ps_compute_unit": {
            "0": {
                "name": "kernel_vcu_decoder:scu_0",
                "base_address": "0",
                "usage": "0",
                "status": "(IDLE)"
            },
...

$xbutil top -d 1
...
###############################################################################

Compute Unit Usage:
CU[@0x17ff100] : 0
CU[@0x1400000] : 0
SCU[@N/A] : 0
SCU[@N/A] : 0
SCU[@N/A] : 0
SCU[@N/A] : 0
SCU[@N/A] : 0
...

$ xbutil -new examine -d 17:00 -r compute-units
-----------------------------------------------------
 Invoking next generation of the xbutil application
-----------------------------------------------------
----------------------------------------------
1/1 [0000:17:00.1] : xilinx_u30_gen3x4_base_1
----------------------------------------------
PL Compute Units
Index   Name                          Base_Address    Usage   Status
0       decoder:decoder_1             0x17ff100       0       (--)
1       verify:verify_1               0x1400000       0       (IDLE)

PS Compute Units
Index   Name                          Base_Address    Usage   Status
0       kernel_vcu_decoder:scu_0      0x0             0       (IDLE)
1       kernel_vcu_decoder:scu_1      0x0             0       (IDLE)
2       kernel_vcu_decoder:scu_2      0x0             0       (IDLE)
3       kernel_vcu_decoder:scu_3      0x0             0       (IDLE)
4       kernel_vcu_decoder:scu_4      0x0             0       (IDLE)


$ xbutil -new examine -d 17:00 -r compute-units -f json
...
"compute_units": [
                {
                    "name": "decoder:decoder_1",
                    "base_address": "0x17ff100",
                    "usage": "0",
                    "type": "PL",
                    "status": {
                        "bit_mask": "0x0"
                    }
                },
                {
                    "name": "verify:verify_1",
                    "base_address": "0x1400000",
                    "usage": "0",
                    "type": "PL",
                    "status": {
                        "bit_mask": "0x4",
                        "bits_set": [
                            "IDLE"
                        ]
                    }
                },
                {
                    "name": "kernel_vcu_decoder:scu_0",
                    "base_address": "0x0",
                    "usage": "0",
                    "type": "PS",
                    "status": {
                        "bit_mask": "0x4",
                        "bits_set": [
                            "IDLE"
                        ]
                    }
                },

```

### CODE REVIEW

File: src/runtime_src/core/tools/common/ReportCu.cpp [hunk 1]
```diff
@@ -112,23 +112,27 @@ populate_cus_new(const xrt_core::device *device)
...
   for (auto& stat : cu_stats) {
     boost::property_tree::ptree ptCu;
     ptCu.put( "name",           stat.name);
     ptCu.put( "base_address",   boost::str(boost::format("0x%x") % stat.base_addr));
     ptCu.put( "usage",          stat.usages);
     ptCu.add_child( std::string("status"),	get_cu_status(stat.status));
-    pt.push_back(std::make_pair("", ptCu));
+    cu_list.push_back(std::make_pair("", ptCu));
   }
+  pt.add_child("compute_units", cu_list);
```
Comment (mamin506) [inline src/runtime_src/core/tools/common/ReportCu.cpp]: Can we named it "pl_compute_units".
Xilinx FPGA user would use terminology "PL" and "PS".
Where PL means Programmable Logic, PS means Processing System.

In this change, we already have "ps_compute_units" for PS kernels. :)

**Review by mamin506 — APPROVED** [review]
Just a suggestion for naming. Others looks good.

File: src/runtime_src/core/tools/common/ReportCu.cpp (hunk 1 above)
Comment (AShivangi) [reply src/runtime_src/core/tools/common/ReportCu.cpp]: Yes, I just updated it. Thanks for the suggestion :)

**Review by larry9523 — APPROVED** [review]

**Review by maxzhen — APPROVED** [review]

**Review by rozumx — CHANGES_REQUESTED** [review]
Lets meet regarding the schema.  It looks like sections are duplicated instead of being merged together.

**Review by rozumx — APPROVED** [review]
Thank you for making the  JSON schema changes.

**Pull request merged**